File: algorithms/easy/minimum_absolute_difference.rs
impl Solution {
    pub fn minimum_abs_difference(mut arr: Vec<i32>) -> Vec<Vec<i32>> {
        let mut ans = Vec::new();
        arr.sort();
        let mut diff = (arr[0] - arr[1]).abs();
        ans.push(vec![arr[0], arr[1]]);
        for i in 1..(arr.len() - 1) {
            let mut t = (arr[i] - arr[i + 1]).abs();
            if diff > t {
                diff = t;
                ans.clear();
                ans.push(vec![arr[i], arr[i + 1]]);
            } else if diff == t {
                ans.push(vec![arr[i], arr[i + 1]]);
            }
        }
        ans
    }
}

File: algorithms/easy/minimum_absolute_difference.kt
class Solution {
  fun minimumAbsDifference(arr: IntArray): List<List<Int>> {
    arr.sort()

    val ans = mutableListOf<List<Int>>()
    var mi = Int.MAX_VALUE

    for (i in 1..<arr.size) {
      val diff = arr[i] - arr[i - 1]
      if (diff < mi) {
        mi = diff
        ans.clear()
        ans.add(listOf(arr[i - 1], arr[i]))
      } else if (diff == mi) {
        ans.add(listOf(arr[i - 1], arr[i]))
      }
    }
    return ans
  }
}
